packages/amplify_datastore_plugin_interface/lib/src/types/query/query_field_operators.dart

```diff
@@ -75,7 +87,8 @@ class EqualQueryOperator<T> extends QueryFieldOperator<T> {
 
   @override
   bool evaluate(T? other) {
-    return other == value;
+    var serializedValue = serializeDynamicValue(value);
```

Comment:
A few callouts here:
- Because `QueryPredicateOperation.evaluate()` (see code below) converts the model to json, `other` will be serialized if it is not a primitive type. `value` needs to be serialized as well to compare non-primitive (i.e. Temporal) types.
- There was a [discussion on the initial observeQuery PR](https://github.com/aws-amplify/amplify-flutter/pull/892/files#r724667952) about alternatives to serializing the entire model to json, which I think is still worth considering. However, it will require codegen changes, and will then be a breaking change.
- While this method is only used for observeQuery(). Query Predicates for query() are evaluated on the native side, and all values are serialized with serializeDynamicValue() prior to being sent to native. This is consistent with how non-primitive types are handled in query().

**QueryPredicateOperation.evaluate()**
https://github.com/aws-amplify/amplify-flutter/blob/8e9b7c5676bc5ac9ae77d102c56d862bbba2a749/packages/amplify_datastore_plugin_interface/lib/src/types/query/query_predicate.dart#L56-L60